File: 7kyu/fizz-buzz-v2/src/main/java/de/klosebrothers/apprentice/fizzbuzz/FizzBuzz.java
package de.klosebrothers.apprentice.fizzbuzz;

public class FizzBuzz {

    public static String fizzBuzz(int number) {
        String result = "";
        result += isModuloZero(number, 3) ? "Fizz" : "";
        result += isModuloZero(number, 5) ? "Buzz" : "";
        return result.isEmpty() ? "" + number : result;
    }

    private static boolean isModuloZero(int number, int modulo) {
        return number % modulo == 0;
    }

    public static void main(String[] args) {
        for (int i = 0; i < 100; i++) {
            System.out.println(fizzBuzz(i + 1));
        }
    }

}

File: Javascript/fizz-buzz-js/fizzbuzz.ts
import { string } from "yargs";

const convertNumToFizzBuzz = (num: number) => {
    let result = num % 3 == 0 ? "Fizz" : "";
    result += num % 5 == 0 ? "Buzz" : "";
    return result === "" ? num.toString() : result;
};

const fizzBuzz = (num: number) => {
    let result = [];
    for (let i = 1; i <= num; i++) {
        result.push(i);
    }
    return result;
};

module.exports = convertNumToFizzBuzz;
